rotkehlchen/tests/api/test_bitcoin.py

```diff
@@ -287,6 +324,75 @@ def test_add_delete_xpub(rotkehlchen_api_server):
     else:
         assert_proper_response(response)
 
+    # Check that addresses that have balances on both bitcoin and bitcoin cash are stored properly
+    with rotki.data.db.conn.read_ctx() as cursor:
+        mutual_in_chain_manager = 0
+        for btc_addr in rotki.chain_manager.accounts.btc:
+            if btc_addr in rotki.chain_manager.accounts.bch:
+                mutual_in_chain_manager += 1
+        # get mutual by querying BCH accounts from known BTC accounts
+        # this checks that bitcoin mappings in the db are not broken
+        mutual_in_db_1 = len(rotki.data.db.get_addresses_to_xpub_mapping(
+            cursor=cursor,
+            blockchain=SupportedBlockchain.BITCOIN,
+            addresses=rotki.chain_manager.accounts.bch,
+        ))
+        # get mutual by querying BTC accounts from known BCH accounts
+        # this checks that bitcoin cash mappings in the db are not broken
+        mutual_in_db_2 = len(rotki.data.db.get_addresses_to_xpub_mapping(
+            cursor=cursor,
+            blockchain=SupportedBlockchain.BITCOIN_CASH,
+            addresses=rotki.chain_manager.accounts.btc,
+        ))
+        # Check that there are some accounts that are used on both btc and bch
+        assert mutual_in_chain_manager > 0
+        # Check that information about these accounts matches between chain manager and the db
+        assert mutual_in_chain_manager == mutual_in_db_1 == mutual_in_db_2
+
+    # Test that editing one xpub doesn't affect other if the only difference is chain
+    json_data_patch = {
+        'xpub': xpub,
+        'label': 'qwerty',
+    }
+    response = requests.patch(api_url_for(
+        rotkehlchen_api_server,
+        'btcxpubresource',
+        blockchain='BTC',
+    ), json=json_data_patch)
+    assert_proper_response(response)
+
+    response = requests.get(api_url_for(
+        rotkehlchen_api_server,
+        'blockchainsaccountsresource',
+        blockchain='BTC',
+    ))
+    if async_query:
+        task_id = assert_ok_async_response(response)
+        result = wait_for_async_task_with_result(rotkehlchen_api_server, task_id, timeout=180)
```

Comment:
why 180? We have constants like `ASYNC_TASK_WAIT_TIMEOUT` and such. Why not use them or a multiple of them?

Reply:
Honestly, I don't remember any reason. Probably appeared here just by an accident.